lib/utils/patterns.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+var RE2;
+var minimatch = require('minimatch');
+try {
+  RE2 = require('re2');
+} catch (er) {
+  RE2 = null;
+}
+
+var patterns = {
+  /**
+   * Evaluates whether a configured redirect/rewrite/custom header should
+   * be applied to a request against a specific path. All three features
+   * are configured with a hash that contains either a Node-like glob path
+   * specification as its `source` or `glob` field, or a RE2 regular
+   * expression as its `regex` field.
+   *
+   * Since Javascript lacks a native library for RE2, Superstatic uses the C
+   * bindings as an optional dependency, and falls over to PCRE if the import
+   * is unavailable. Under most circumstances not involving named capturing
+   * groups, the two libraries should have identical behavior.
+   *
+   * No special consideration is taken if the configuration hash contains both
+   * a glob and a regex. normalizeConfig() will error in that case.
+   *
+   * @param {string} path The URL path from the request.
+   * @param {Object} config A dictionary from a sanitized JSON configuration.
+   * @return {boolean} Whether the config should be applied to the request.
+   */
+  configMatcher: function(path, config) {
+    var glob = config.glob || config.source;
+    var regex = config.regex;
+    if (glob && minimatch(path, glob)) {
+      return true;
+    }
+    if (regex && RE2) {
```

Comment:
Why not:

```js
if (regex) {
  path.match(createRaw(regex)) !== null
}
```

instead of having extra conditionals

Reply:
that is an extraordinarily good question